dhcpv4/nclient4/client.go

```diff
@@ -193,21 +249,19 @@ func (c *Client) Close() error {
 	return err
 }
 
-func isErrClosing(err error) bool {
-	// Unfortunately, the epoll-connection-closed error is internal to the
-	// net library.
-	return strings.Contains(err.Error(), "use of closed network connection")
+func (c *Client) isClosed() bool {
+	return atomic.LoadUint32(&c.closed) != 0
```

Comment:
closed is never set? Also, why the change?

Reply:
`c.closed` is set in `c.Close()`. The change because the previous way is:

* Hacky. Search a substring in the error description is an unstrict/risky way to handle errors.
* Wrong. I can implement `net.PacketConn` which returns other error. And this package accepts arbitrary `net.PacketConn`-s via `NewWithConn(...)`.